api/utils/taskmanager.js

```diff
@@ -657,4 +659,43 @@ taskmanager.rerunTask = function(options, callback) {
         }
     });
 };
+    
+    /**
+ *  Create a callback for getting result, including checking gridfs
+ *  @param {function} callback - callback for the result
+ *  @param {object} options - options object
+ *  @returns {function} callback to use for db query
+ */
+function getResult(callback,options) {
+    return function(err, data) {
+        if (!err) {
+            if (data && options && options.subtask_key && data.taskgroup === true) {
+                taskmanager.getResultByQuery({db: options.db, query: {subtask: data._id, subtask_key: options.subtask_key}}, getResult(function(err2, subtask){
+                    if (!subtask) {
+                        taskmanager.rerunTask({db: options.db, id: data._id}, function() {});
+                    }
+                    callback(err2, subtask);
+                }));
+            }
+            else if (data && data.gridfs) {
+                countlyFs.gridfs.getData("task_results", data._id + "", {id: data._id}, function(err2, largeData) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['countlyFs' is not defined.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5125857)